anchore_engine/common/helpers.py

```diff
@@ -29,19 +29,24 @@ def make_response_error(errmsg, in_httpcode=None, details=None):
 
     if isinstance(errmsg, Exception):
         if 'anchore_error_json' in errmsg.__dict__:
-            if {'message', 'httpcode', 'detail'}.issubset(set(errmsg.__dict__['anchore_error_json'])):
-                ret.update(errmsg.__dict__['anchore_error_json'])
+            # Try to load it as json
+            try:
+                err_json = json.loads(errmsg.__dict__.get('anchore_error_json', None))
+            except ValueError:
```

Comment:
@dakaneye, I'm seeing this fail in a lot of because anchore_error_json is a dict. `json.loads()` can only be used for loading a str, bytes or bytearray types. Is there a case where anchore_error_json is anything but a dict? 

Another issue here is the limited exception handling. In the case where `json.loads()` fails because the input is not one of the above types, the library throws a TypeError. That's not caught here. The end result to the user is an error handling an internal exception, not the actual error itself. This is one of the reasons why I have used broader catch clauses

Reply:
This should be in its own ticket now that the change is merged.

Reply:
Will open and fix.